src/ResourceManager/Profile/Commands.Profile/Environment/AddAzureRMEnvironment.cs

```diff
@@ -113,6 +122,21 @@ public override void ExecuteCmdlet()
         {
             var profileClient = new RMProfileClient(AzureRmProfileProvider.Instance.Profile);
 
+            if (this.ParameterSetName.Equals(ArmParameterSet, StringComparison.Ordinal))
+            {
+                MetadataResponse metadataEndpoints = null;
+                metadataEndpoints = EnvironmentHelper.RetrieveMetaDataEndpoints(ResourceManagerEndpoint).Result;
+                string domain = EnvironmentHelper.RetrieveDomain(metadataEndpoints.PortalEndpoint);
+                ActiveDirectoryEndpoint = metadataEndpoints.authentication.LoginEndpoint.TrimEnd('/') + '/';
+                ActiveDirectoryServiceEndpointResourceId = metadataEndpoints.authentication.Audiences[0];
+                GalleryEndpoint = metadataEndpoints.GalleryEndpoint;
+                GraphEndpoint = metadataEndpoints.GraphEndpoint;
+                AzureKeyVaultDnsSuffix = $"vault.{domain}".ToLowerInvariant();
+                AzureKeyVaultServiceEndpointResourceId = ($"https://vault.{domain}".ToLowerInvariant());
+                StorageEndpoint = domain;
```

Comment:
No, but I guess we agreed upon adding this for stack-dev branch for now. We will have to revisit during stack-dev to dev branch merge.